internal/testutils/stats/test_metrics_recorder.go

```diff
@@ -0,0 +1,211 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package stats implements a TestMetricsRecorder utility.
+package stats
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	estats "google.golang.org/grpc/experimental/stats"
+	"google.golang.org/grpc/internal/testutils"
+	"google.golang.org/grpc/stats"
+)
+
+// TestMetricsRecorder is a MetricsRecorder to be used in tests. It sends
+// recording events on channels and provides helpers to check if certain events
+// have taken place. It also persists metrics data keyed on the metrics
+// descriptor.
+type TestMetricsRecorder struct {
+	t *testing.T
+
+	// TODO: scale this persisted storage up with label key values.
```

Comment:
Sorry, I'm not sure what this means?

Reply:
Discussed offline; decided to get rid of these maps entirely and add these when needed.